### PR TITLE
Clean up associated types in ServerModule

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: fedimint-api/src/module/mod.rs [hunk 1]
```diff
@@ -425,8 +425,7 @@ where
 
 #[async_trait]
 pub trait ServerModule: Debug + Sized {
-    const KIND: ModuleKind;
-
+    type Gen: ModuleGen;
```
Comment (elsirion) [inline fedimint-api/src/module/mod.rs:425]: Now I'm unsure which way around this reference makes the most sense … but I think this should work (just changes if you attach modules or module gens to the federation builder eventually).

File: fedimint-api/src/module/mod.rs [hunk 2]
```diff
@@ -427,10 +424,6 @@ where
 pub trait ServerModule: Debug + Sized {
     type Gen: ModuleGen;
     type Decoder: Decoder;
-    type Input: Input;
-    type Output: Output;
-    type OutputOutcome: OutputOutcome;
-    type ConsensusItem: ModuleConsensusItem;
```
Comment (elsirion) [inline fedimint-api/src/module/mod.rs:434]: :+1: discussed that with @mxz42 a while ago

**Review by elsirion — APPROVED** [review]

**Review by mxz42 — APPROVED** [review]
nice!

File: fedimint-api/src/module/mod.rs (hunk 1 above)
Comment (dpc) [reply fedimint-api/src/module/mod.rs:425]: Might be both.

**Pull request merged**